marconi-chain-index/src/Marconi/ChainIndex/Experimental/Indexers/Worker.hs

```diff
@@ -20,50 +20,84 @@ import Data.Text (Text)
   , catchupConfig :: Core.CatchupConfig
   , eventExtractor :: input -> m (Maybe event)
   , logger :: Trace m (Core.IndexerEvent C.ChainPoint)
   }
 
+data StandardWorker m input event indexer = StandardWorker
+  { standardWorkerResumePoints :: ![Core.Point event]
+  , standardWorkerIndexerVar :: !(MVar (StandardIndexer m indexer event))
+  , standardWorker :: Core.WorkerM m (WithDistance input) (Core.Point event)
+  }
+
+instance Core.OrdPoint C.ChainPoint where
+  comparePoint C.ChainPointAtGenesis C.ChainPointAtGenesis = Core.Same
+  comparePoint C.ChainPoint{} C.ChainPointAtGenesis = Core.After
+  comparePoint C.ChainPointAtGenesis C.ChainPoint{} = Core.Before
+  comparePoint (C.ChainPoint p1 h1) (C.ChainPoint p2 h2)
+    | p1 < p2 = Core.Before
+    | p1 > p2 = Core.After
+    | p1 == p2 && h1 == h2 = Core.Same
+    | otherwise = Core.Fork
+
 -- | Create a worker for the given indexer with some standard catchup values
-catchupWorker
+mkStandardWorker
   :: ( MonadIO m
      , MonadIO n
-     , Core.WorkerIndexer (ExceptT Core.IndexerError m) event indexer
+     , Core.WorkerIndexerType (ExceptT Core.IndexerError m) event indexer
      , Core.Point event ~ C.ChainPoint
+     , Core.IsSync n event indexer
      )
   => StandardWorkerConfig m input event
   -> indexer event
-  -> n (MVar (StandardIndexer m indexer event), Core.WorkerM m (WithDistance input) (Core.Point event))
-catchupWorker config = catchupWorkerWithFilter config Just
+  -> n (StandardWorker m input event indexer)
+mkStandardWorker config = mkStandardWorkerWithFilter config Just
 
 -- | Create a worker for the given indexer with some standard catchup values with extra filtering
-catchupWorkerWithFilter
+mkStandardWorkerWithFilter
   :: ( MonadIO n
      , MonadIO m
-     , Core.WorkerIndexer (ExceptT Core.IndexerError m) event indexer
+     , Core.WorkerIndexerType (ExceptT Core.IndexerError m) event indexer
+     , Core.IsSync n event indexer
+     , Ord (Core.Point event)
      , Core.Point event ~ C.ChainPoint
      )
   => StandardWorkerConfig m input event
   -> (event -> Maybe event)
   -> indexer event
-  -> n (MVar (StandardIndexer m indexer event), Core.WorkerM m (WithDistance input) (Core.Point event))
-catchupWorkerWithFilter config eventFilter indexer =
+  -> n (StandardWorker m input event indexer)
+mkStandardWorkerWithFilter config eventFilter indexer =
   let chainPointDistance :: Core.Point (WithDistance a) -> WithDistance a -> Word64
       chainPointDistance _ = Distance.chainDistance
       mapEventUnderDistance = fmap sequence . traverse (eventExtractor config)
-   in Core.createWorker (workerName config) mapEventUnderDistance $
-        Core.withTrace (logger config) $
-          Core.withCatchup chainPointDistance (catchupConfig config) $
-            Core.withTransform id (eventFilter . Distance.getEvent) indexer
+   in do
+        -- TODO Change withResume so that we can pass in chain points directly instead of using
+        -- function.
```

Comment:
What do you think? Should I just do it here?

Reply:
now that we have the possibility to get it from `IsSync`, can we even use this directly? (but we can pass the points as well)